src/main/java/com/stuypulse/robot/commands/swerve/SwerveDriveToShoot.java

```diff
@@ -0,0 +1,113 @@
+package com.stuypulse.robot.commands.swerve;
+
+import com.stuypulse.robot.constants.Settings.Alignment;
+import com.stuypulse.robot.constants.Settings.Alignment.Rotation;
+import com.stuypulse.robot.constants.Settings.Alignment.Translation;
+import com.stuypulse.robot.subsystems.odometry.Odometry;
+import com.stuypulse.robot.subsystems.swerve.SwerveDrive;
+import com.stuypulse.robot.util.HolonomicController;
+import com.stuypulse.stuylib.control.angle.feedback.AnglePIDController;
+import com.stuypulse.stuylib.control.feedback.PIDController;
+import com.stuypulse.stuylib.math.Angle;
+import com.stuypulse.stuylib.math.Vector2D;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.smartdashboard.FieldObject2d;
+import edu.wpi.first.wpilibj2.command.Command;
+
+public class SwerveDriveToShoot extends Command {
+    /*
+     * swerve
+     * holonomi controller
+     * targetpose pose2d
+     * targetPose2d as a field object 2d
+     * 
+     * contrusctor (init them, add requirements)
+     * initialize (do the math to get the target pose)
+     * execute (set the target pose to the controller and then setChassisSpeeds to it)
+     * isFinished (check if the controller is done within the tolerances)
+     * end (stop the swerve)    
+     * 
+     */
+    private final SwerveDrive swerve;
+    private Pose2d targetPose;
+    private final HolonomicController controller;
+    private final FieldObject2d targetPose2d;
+
+    public SwerveDriveToShoot(Pose2d targetPose) {
```

Comment:
Remove unused targetPose param

Reply:
This command is supposed to align from any point on the field to a certain shooting distance away from the speaker and to point at the speaker target position. then shoot.